gsy_framework/sim_results/carbon_emissions/results.py

```diff
@@ -42,31 +42,40 @@ def _find_start_and_end_dates(self, dates: List) -> Tuple[DateTime, DateTime]:
 
     def _create_hourly_timestamps(self, start: DateTime, end: DateTime) -> List[DateTime]:
         """Generate a list of datetime objects for each hour between start and end"""
+        if not isinstance(start, pendulum.DateTime):
+            start = pendulum.instance(start)
+        if not isinstance(end, pendulum.DateTime):
+            end = pendulum.instance(end)
         return [start.add(hours=hour) for hour in range(int((end - start).total_hours()) + 1)]
 
     def query_carbon_ratio_from_static_source(
-        self, country_code: str, start: DateTime, end: DateTime
+        self, country_code: str, start: DateTime, end: DateTime, file_prefix=""
     ) -> List[Dict]:
         """Calculate the total carbon generated based on the specified statistic"""
 
+        if not file_prefix:
+            file_prefix = "gsy_framework"
```

Comment:
uff, I am not a fan of this solution unfortunately. 
How about doing something I did in my latest gsy-web PR:
1. create a variable that holds the path to the resources https://github.com/gridsingularity/gsy-web/pull/2285/files#diff-d37ae1bfda98224e3012b3ce64f0b49bf6e7ecf1d3016645a6e5bb85c05e4e13R3
2. re-use it by importing when opening a file: https://github.com/gridsingularity/gsy-web/pull/2285/files#diff-d43286a340dfb891017855642cac303271c1c8aa3318652e5d96d046fda405f3R154

Reply:
Fine for here, and let me adapt it. However this solution will not work when importing gsy-framework in another repo (e.g. gsy-web) because this file ends up in the site-packages directory, therefore we need to refer it from the imported package and not from the project that imports it. 
Let me change it here though so that it is resolved.

Reply:
Tested it and you were right, the relative path of the current file worked! Thanks for the suggestion

Reply:
> Fine for here, and let me adapt it. However this solution will not work when importing gsy-framework in another repo (e.g. gsy-web) because this file ends up in the site-packages directory, therefore we need to refer it from the imported package and not from the project that imports it.

I understood, that my suggestion saves the absolute path of the static file directory to the STATIC_TEST_FILE_DIR when initialising. So, when you import it from another repo, this should not be untouched and point to the site-packages folder where it lives, or did I get it wrong?

Reply:
Yes, but this is what I do not understand:
> that my suggestion saves the absolute path of the static file directory to the STATIC_TEST_FILE_DIR when initialising

I assume that by "static file directory" you mean the `gsy_framework/resources/` path?

Reply:
yes, sorry, I should not have referred to my but this situation here.